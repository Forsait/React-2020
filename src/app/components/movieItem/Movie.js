import React from 'react';
import Link from 'next/link'

import { getYear } from '../../utils/date';
import styles from './Movie.module.scss';

export default function Movie(props) {
  return (
    <div className={styles.film_item}>
      <img className={styles.film_poster} src={props.data.poster_path}/>
      <div className={styles.movie_info}>
        <Link href="/movieinfo/[id]" as={`/movieinfo/${props.data.id}`}>
          <a>{props.data.title}</a>
        </Link>
        <span className={styles.movie_date}>{getYear(props.data.release_date)}</span>
      </div>
      <div className={styles.genre}>{props.data.genres[0]}</div>
    </div>
  )
}