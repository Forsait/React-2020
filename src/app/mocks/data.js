export const film = {
  budget: 0,
  genres: ['Action', 'Adventure', 'Drama', 'Science Fiction', 'Thriller'],
  id: 351286,
  overview: 'A volcanic eruption threatens',
  poster_path: 'https://image.tmdb.org/t/p/w500/ln6d5Okr6VK5vfQVobJTiYxeD0l.jpg',
  release_date: '2018-06-01',
  revenue: 0,
  runtime: null,
  tagline: 'Life finds a way',
  title: 'Jurassic World: Fallen Kingdom',
  vote_average: 0,
  vote_count: 28,
};

export const moviesData = {
  data: new Array(6).fill(film),
  limit: 0,
  offset: 0,
  total: 0,
};

export const radioOptions = {
  title: 'Radio title',
  list: [
    {
      key: 'key 1',
      title: 'title 1',
    },
    {
      key: 'key 2',
      title: 'title 2',
    },
  ],
};
