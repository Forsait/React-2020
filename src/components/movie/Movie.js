import React from 'react';

import styles from './Movie.module.scss';

export default function Movie(props) {
  const releaseDate = props.data.release_date.slice(0,4);

  return (
    <div className={styles.film_item}>
      <img className={styles.film_poster} src={props.data.poster_path}/>
      <div className={styles.movie_info}>
        <span>{props.data.title}</span>
        <span className={styles.movie_date}>{releaseDate}</span>
      </div>
      <div className={styles.genre}>{props.data.genres[0]}</div>
    </div>
  )
}