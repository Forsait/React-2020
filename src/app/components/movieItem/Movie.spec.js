import React from 'react';
import { shallow } from 'enzyme';

import Movie from './Movie';
import { film } from '../../mocks/data';

describe('MovieItem component', () => {

  it('should be render correctly', () => {
    const com = shallow(<Movie data={film} />);
    expect(com).toMatchSnapshot()
  });

});
