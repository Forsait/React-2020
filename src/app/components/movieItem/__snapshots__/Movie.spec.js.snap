// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieItem component should be render correctly 1`] = `
<div>
  <img
    src="https://image.tmdb.org/t/p/w500/ln6d5Okr6VK5vfQVobJTiYxeD0l.jpg"
  />
  <div>
    <span>
      Jurassic World: Fallen Kingdom
    </span>
    <span>
      2018
    </span>
  </div>
  <div>
    Action
  </div>
</div>
`;
